plugins/obs-filters/color-grade-filter.c

```diff
@@ -28,6 +32,54 @@ static const char *color_grade_filter_get_name(void *unused)
 	return obs_module_text("ColorGradeFilter");
 }
 
+static gs_texture_t *make_clut_texture(const enum gs_color_format format,
+				       const uint32_t image_width,
+				       const uint32_t image_height,
+				       const uint8_t *data)
+{
+	if (image_width % LUT_WIDTH != 0)
+		return NULL;
+
+	if (image_height % LUT_WIDTH != 0)
+		return NULL;
+
+	const uint32_t pixel_count = LUT_WIDTH * LUT_WIDTH * LUT_WIDTH;
+	if ((image_width * image_height) != pixel_count)
```

Comment:
This breaks lookup tables that aren't exactly 64x64x64, such as a LUT that is 256x256x256 and would cover the entire RGB color spectrum.

Reply:
LUT images that could work if supported:

* 1x1x1: Basically useless.
* 4x4x4: 2D Size 8x8. Useful for quick and dirty grayscale, interpolation makes up the majority of quality. Very very fast for LUT.
* 16x16x16: 2D Size 64x64. Higher quality Grayscale and isolation at the cost of more sampling.
* 64x64x64: The current size of 512x512, covers 1.56% of the representable spectrum, making up the rest of interpolation.
* 256x256x256: 4096x4096, **covers the entire 8-bit color range**.
* 1024x1024x1024: 32768x32768, **covers the entire 10-bit color range**, will definitely require volume textures as the U/V limits for 2D textures cap out at 16384 - same for OpenGL it seems. 3D textures can go as big as 2048x2048x2048, which would cover 11-bits, but nothing uses 11-bits.

Reply:
LUTs that aren't 64x64x64 were not previously supported. The shader was hard-coded for them.

Reply:
Yeah, I realized in my 2nd comment.